tests/src/androidTest/java/com/squareup/duktape/DuktapeTest.java

```diff
@@ -47,7 +47,31 @@
       duktape.evaluate("nope();");
       fail();
     } catch (DuktapeException e) {
-      assertThat(e).hasMessage("ReferenceError: identifier 'nope' undefined");
+      assertThat(e.getMessage()).startsWith("ReferenceError: identifier 'nope' undefined");
+    }
+  }
+
+  @Test public void exceptionsInScriptIncludeStackTrace() {
+    try {
+      duktape.evaluate("\n"
+            + "f1();\n"           // Line 2.
+            + "\n"
+            + "function f1() {\n"
+            + "  f2();\n"         // Line 5.
+            + "}\n"
+            + "\n"
+            + "\n"
+            + "function f2() {\n"
+            + "  nope();\n"       // Line 10.
+            + "}\n");
+      fail();
+    } catch (DuktapeException e) {
+      // The first line is the error type and message.
+      assertThat(e.getMessage()).startsWith("ReferenceError: identifier 'nope' undefined");
+      // Each following line in the stacktrace is <function> <filename>:<linenumber>.
+      assertThat(e.getMessage()).containsMatch("f2 .*:10");
```

Comment:
Oh! That's unusual. I'll have a look into that upstream.

On Sun, Oct 25, 2015, 12:54 PM Shawn Zurbrigg <notifications@github.com>
wrote:

> In tests/src/androidTest/java/com/squareup/duktape/DuktapeTest.java
> <https://github.com/square/duktape-android/pull/22#discussion_r42947063>:
>
> > +            + "f1();\n"           // Line 2.
> > +            + "\n"
> > +            + "function f1() {\n"
> > +            + "  f2();\n"         // Line 5.
> > +            + "}\n"
> > +            + "\n"
> > +            + "\n"
> > +            + "function f2() {\n"
> > +            + "  nope();\n"       // Line 10.
> > +            + "}\n");
> > +      fail();
> > +    } catch (DuktapeException e) {
> > +      // The first line is the error type and message.
> > +      assertThat(e.getMessage()).startsWith("ReferenceError: identifier 'nope' undefined");
> > +      // Each following line in the stacktrace is <function> <filename>:<linenumber>.
> > +      assertThat(e.getMessage()).containsMatch("f2 .*:10");
>
> I see, thanks. But containsMatch() returns void, unless I have something
> else messed up?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/duktape-android/pull/22/files#r42947063>.
>